python-models/server/server.py

```
@@ -0,0 +1,223 @@
+        row = {col: getattr(features, col) for col in FEATURE_COLS}
+        X = pd.DataFrame([row], columns=FEATURE_COLS)
+
+        outputs: Dict[str, int] = {}
+        for target, pipe in models.items():
+            # Each saved pipeline includes preprocessing and model
+            pred_value = float(pipe.predict(X)[0])
+            outputs[target] = max(0, int(round(pred_value)))
+
+        return outputs
+    except Exception as e:  # noqa: BLE001
+        raise HTTPException(status_code=500, detail=str(e))
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)